### PR TITLE
pmd-ndctl.go: Add start-time check for writable /sys

### PR DESCRIPTION
**Pull request by okartau**
In Devicemode_direct (alias ndctl), the driver can operate
only if /sys is mounted in read-write mode.
It has been seen however that sometimes container gets
deployed in such a way that /sys is mounted read-only.
It makes sense to detect writable /sys at start-time
and refuse to run, instead of stumbling to that later in operation.

### CODE REVIEW

**Comment by okartau** [comment]
This implements one of two checks proposed in #172

**Comment by okartau** [comment]
For Devicemode:LVM, we could add similar check in startup phase of pmem-ns-init which needs writable /sys in LVM mode. But, two points suggest we can perhaps skip adding:
- pmem-ns-init runs as initContainer, and I haven't yet seen case when /sys is not writable there
- if we still hit read-only /sys, initContainer running before actual driver, either:
   - gets "read-only system" error at namespace creation attempt and exits (clear enough indication), 
   - or, it does not need to create namespaces (set of namespaces exists) and plugin is able to operate in LVM mode on top of existing set of namespaces.

so I think in LVM mode we can live without such start-up stage check

File: pkg/pmem-device-manager/pmd-ndctl.go [hunk 1]
```diff
@@ -19,6 +20,24 @@ func NewPmemDeviceManagerNdctl() (PmemDeviceManager, error) {
...
+	// Check is /sys writable. If not then there is no point starting
+	mounter := mount.New("")
+	mounts, err := mounter.List()
+	for i := range mounts {
+		glog.Infof("NewPmemDeviceManagerNdctl: Check mounts: device=%s path=%s opts=%s",
+			mounts[i].Device, mounts[i].Path, mounts[i].Opts)
+		if mounts[i].Device == "sysfs" && mounts[i].Path == "/sys" {
+			for _, opt := range mounts[i].Opts {
+				if opt == "rw" {
+					glog.Infof("NewPmemDeviceManagerNdctl: /sys mounted read-write, good")
```
Comment (avalluri) [inline pkg/pmem-device-manager/pmd-ndctl.go]: can't we return from here.

**Comment by okartau** [comment]
the log in "rw" case is bit of in extra-for-debug-only style, as is also the "check mounts:" trace log above. In final version those two logs could both go away. But do we delete those right here or later, our decision.

**Comment by okartau** [comment]
Having few log lines once during startup is not too noisy, could provide valuable info about mounts from inside container, so lets keep them and merge.

**Pull request merged**